src/privileged/api.js

```diff
@@ -19,7 +19,7 @@ this.fxmonitor = class extends ExtensionAPI {
   }
 
   onShutdown(shutdownReason) {
-    if (!FirefoxMonitorContainer.FirefoxMonitor) {
+    if (Services.startup.shuttingDown || !FirefoxMonitorContainer.FirefoxMonitor) {
```

Comment:
sorry for the drive-by, you can also just check if `shutdownReason == "APP_SHUTDOWN"`

Reply:
I purposely did not do this, because, for example, if the addon is installed as a temporary add-on, the reason is ADDON_UNINSTALLED. And we *do* want to handle that case if we're not shutting down.

Reply:
It's also ADDON_UNINSTALLED (or disabled) if there is a pending disabled.

In the real world since this is being shipped as a system add-on, this shouldn't matter, but for hygiene and resilience to shipping strategy I prefer to just use Services.startup.shuttingDown here.